// inversify-logger-middleware/dist/index.js
/**
 * inversify-logger-middleware v.1.0.1 - A basic logger middleware for InversifyJS
 * Copyright (c) 2015 Remo H. Jansen <remo.jansen@wolksoftware.com> (http://www.remojansen.com)
 * MIT inversify.io/LICENSE
 * https://github.com/inversify/inversify-logger-middleware#readme
 */
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.inversifyLoggerMidleware = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
'use strict';
module.exports = function () {
	return /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
};

},{}],2:[function(require,module,exports){
'use strict';

function assembleStyles () {
	var styles = {
		modifiers: {
			reset: [0, 0],
			bold: [1, 22], // 21 isn't widely supported and 22 does the same thing
			dim: [2, 22],
			italic: [3, 23],
			underline: [4, 24],
			inverse: [7, 27],
			hidden: [8, 28],
			strikethrough: [9, 29]
		},
		colors: {
			black: [30, 39],
			red: [31, 39],
			green: [32, 39],
			yellow: [33, 39],
			blue: [34, 39],
			magenta: [35, 39],
			cyan: [36, 39],
			white: [37, 39],
			gray: [90, 39]
		},
		bgColors: {
			bgBlack: [40, 49],
			bgRed: [41, 49],
			bgGreen: [42, 49],
			bgYellow: [43, 49],
			bgBlue: [44, 49],
			bgMagenta: [45, 49],
			bgCyan: [46, 49],
			bgWhite: [47, 49]
		}
	};

	// fix humans
	styles.colors.grey = styles.colors.gray;

	Object.keys(styles).forEach(function (groupName) {
		var group = styles[groupName];

		Object.keys(group).forEach(function (styleName) {
			var style = group[styleName];

			styles[styleName] = group[styleName] = {
				open: '\u001b[' + style[0] + 'm',
				close: '\u001b[' + style[1] + 'm'
			};
		});

		Object.defineProperty(styles, groupName, {
			value: group,
			enumerable: false
		});
	});

	return styles;
}

Object.defineProperty(module, 'exports', {
	enumerable: true,
	get: assembleStyles
});

},{}],3:[function(require,module,exports){
(function (process){
'use strict';
var escapeStringRegexp = require('escape-string-regexp');
var ansiStyles = require('ansi-styles');
var stripAnsi = require('strip-ansi');
var hasAnsi = require('has-ansi');
var supportsColor = require('supports-color');
var defineProps = Object.defineProperties;
var isSimpleWindowsTerm = process.platform === 'win32' && !/^xterm/i.test(process.env.TERM);

function Chalk(options) {
	// detect mode if not set manually
	this.enabled = !options || options.enabled === undefined ? supportsColor : options.enabled;
}

// use bright blue on Windows as the normal blue color is illegible
if (isSimpleWindowsTerm) {
	ansiStyles.blue.open = '\u001b[94m';
}

var styles = (function () {
	var ret = {};

	Object.keys(ansiStyles).forEach(function (key) {
		ansiStyles[key].closeRe = new RegExp(escapeStringRegexp(ansiStyles[key].close), 'g');

		ret[key] = {
			get: function () {
				return build.call(this, this._styles.concat(key));
			}
		};
	});

	return ret;
})();

var proto = defineProps(function chalk() {}, styles);

function build(_styles) {
	var builder = function () {
		return applyStyle.apply(builder, arguments);
	};

	builder._styles = _styles;
	builder.enabled = this.enabled;
	// __proto__ is used because we must return a function, but there is
	// no way to create a function with a different prototype.
	/* eslint-disable no-proto */
	builder.__proto__ = proto;

	return builder;
}

function applyStyle() {
	// support varags, but simply cast to string in case there's only one arg
	var args = arguments;
	var argsLen = args.length;
	var str = argsLen !== 0 && String(arguments[0]);

	if (argsLen > 1) {
		// don't slice `arguments`, it prevents v8 optimizations
		for (var a = 1; a < argsLen; a++) {
			str += ' ' + args[a];
		}
	}

	if (!this.enabled || !str) {
		return str;
	}

	var nestedStyles = this._styles;
	var i = nestedStyles.length;

	// Turns out that on Windows dimmed gray text becomes invisible in cmd.exe,
	// see https://github.com/chalk/chalk/issues/58
	// If we're on Windows and we're dealing with a gray color, temporarily make 'dim' a noop.
	var originalDim = ansiStyles.dim.open;
	if (isSimpleWindowsTerm && (nestedStyles.indexOf('gray') !== -1 || nestedStyles.indexOf('grey') !== -1)) {
		ansiStyles.dim.open = '';
	}

	while (i--) {
		var code = ansiStyles[nestedStyles[i]];

		// Replace any instances already present with a re-opening code
		// otherwise only the part of the string until said closing code
		// will be colored, and the rest will simply be 'plain'.
		str = code.open + str.replace(code.closeRe, code.open) + code.close;
	}

	// Reset the original 'dim' if we changed it to work around the Windows dimmed gray issue.
	ansiStyles.dim.open = originalDim;

	return str;
}

function init() {
	var ret = {};

	Object.keys(styles).forEach(function (name) {
		ret[name] = {
			get: function () {
				return build.call(this, [name]);
			}
		};
	});

	return ret;
}

defineProps(Chalk.prototype, init());

module.exports = new Chalk();
module.exports.styles = ansiStyles;
module.exports.hasColor = hasAnsi;
module.exports.stripColor = stripAnsi;
module.exports.supportsColor = supportsColor;

}).call(this,require('_process'))

},{"_process":6,"ansi-styles":2,"escape-string-regexp":4,"has-ansi":5,"strip-ansi":7,"supports-color":8}],4:[function(require,module,exports){
'use strict';

var matchOperatorsRe = /[|\\{}()[\]^$+*?.]/g;

module.exports = function (str) {
	if (typeof str !== 'string') {
		throw new TypeError('Expected a string');
	}

	return str.replace(matchOperatorsRe, '\\$&');
};

},{}],5:[function(require,module,exports){
'use strict';
var ansiRegex = require('ansi-regex');
var re = new RegExp(ansiRegex().source); // remove the `g` flag
module.exports = re.test.bind(re);

},{"ansi-regex":1}],6:[function(require,module,exports){
// shim for using process in browser
var process = module.exports = {};

// cached from whatever global is present so that test runners that stub it
// don't break things.  But we need to wrap it in a try catch in case it is
// wrapped in strict mode code which doesn't define any globals.  It's inside a
// function because try/catches deoptimize in certain engines.

var cachedSetTimeout;
var cachedClearTimeout;

(function () {
    try {
        cachedSetTimeout = setTimeout;
    } catch (e) {
        cachedSetTimeout = function () {
            throw new Error('setTimeout is not defined');
        }
    }
    try {
        cachedClearTimeout = clearTimeout;
    } catch (e) {
        cachedClearTimeout = function () {
            throw new Error('clearTimeout is not defined');
        }
    }
} ())
function runTimeout(fun) {
    if (cachedSetTimeout === setTimeout) {
        //normal enviroments in sane situations
        return setTimeout(fun, 0);
    }
    try {
        // when when somebody has screwed with setTimeout but no I.E. maddness
        return cachedSetTimeout(fun, 0);
    } catch(e){
        try {
            // When we are in I.E. but the script has been evaled so I.E. doesn't trust the global object when called normally
            return cachedSetTimeout.call(null, fun, 0);
        } catch(e){
            // same as above but when it's a version of I.E. that must have the global object for 'this', hopfully our context correct otherwise it will throw a global error
            return cachedSetTimeout.call(this, fun, 0);
        }
    }


}
function runClearTimeout(marker) {
    if (cachedClearTimeout === clearTimeout) {
        //normal enviroments in sane situations
        return clearTimeout(marker);
    }
    try {
        // when when somebody has screwed with setTimeout but no I.E. maddness
        return cachedClearTimeout(marker);
    } catch (e){
        try {
            // When we are in I.E. but the script has been evaled so I.E. doesn't  trust the global object when called normally
            return cachedClearTimeout.call(null, marker);
        } catch (e){
            // same as above but when it's a version of I.E. that must have the global object for 'this', hopfully our context correct otherwise it will throw a global error.
            // Some versions of I.E. have different rules for clearTimeout vs setTimeout
            return cachedClearTimeout.call(this, marker);
        }
    }



}
var queue = [];
var draining = false;
var currentQueue;
var queueIndex = -1;

function cleanUpNextTick() {
    if (!draining || !currentQueue) {
        return;
    }
    draining = false;
    if (currentQueue.length) {
        queue = currentQueue.concat(queue);
    } else {
        queueIndex = -1;
    }
    if (queue.length) {
        drainQueue();
    }
}

function drainQueue() {
    if (draining) {
        return;
    }
    var timeout = runTimeout(cleanUpNextTick);
    draining = true;

    var len = queue.length;
    while(len) {
        currentQueue = queue;
        queue = [];
        while (++queueIndex < len) {
            if (currentQueue) {
                currentQueue[queueIndex].run();
            }
        }
        queueIndex = -1;
        len = queue.length;
    }
    currentQueue = null;
    draining = false;
    runClearTimeout(timeout);
}

process.nextTick = function (fun) {
    var args = new Array(arguments.length - 1);
    if (arguments.length > 1) {
        for (var i = 1; i < arguments.length; i++) {
            args[i - 1] = arguments[i];
        }
    }
    queue.push(new Item(fun, args));
    if (queue.length === 1 && !draining) {
        runTimeout(drainQueue);
    }
};

// v8 likes predictible objects
function Item(fun, array) {
    this.fun = fun;
    this.array = array;
}
Item.prototype.run = function () {
    this.fun.apply(null, this.array);
};
process.title = 'browser';
process.browser = true;
process.env = {};
process.argv = [];
process.version = ''; // empty string to avoid regexp issues
process.versions = {};

function noop() {}

process.on = noop;
process.addListener = noop;
process.once = noop;
process.off = noop;
process.removeListener = noop;
process.removeAllListeners = noop;
process.emit = noop;

process.binding = function (name) {
    throw new Error('process.binding is not supported');
};

process.cwd = function () { return '/' };
process.chdir = function (dir) {
    throw new Error('process.chdir is not supported');
};
process.umask = function() { return 0; };

},{}],7:[function(require,module,exports){
'use strict';
var ansiRegex = require('ansi-regex')();

module.exports = function (str) {
	return typeof str === 'string' ? str.replace(ansiRegex, '') : str;
};

},{"ansi-regex":1}],8:[function(require,module,exports){
(function (process){
'use strict';
var argv = process.argv;

var terminator = argv.indexOf('--');
var hasFlag = function (flag) {
	flag = '--' + flag;
	var pos = argv.indexOf(flag);
	return pos !== -1 && (terminator !== -1 ? pos < terminator : true);
};

module.exports = (function () {
	if ('FORCE_COLOR' in process.env) {
		return true;
	}

	if (hasFlag('no-color') ||
		hasFlag('no-colors') ||
		hasFlag('color=false')) {
		return false;
	}

	if (hasFlag('color') ||
		hasFlag('colors') ||
		hasFlag('color=true') ||
		hasFlag('color=always')) {
		return true;
	}

	if (process.stdout && !process.stdout.isTTY) {
		return false;
	}

	if (process.platform === 'win32') {
		return true;
	}

	if ('COLORTERM' in process.env) {
		return true;
	}

	if (process.env.TERM === 'dumb') {
		return false;
	}

	if (/^screen|^xterm|^vt100|color|ansi|cygwin|linux/i.test(process.env.TERM)) {
		return true;
	}

	return false;
})();

}).call(this,require('_process'))

},{"_process":6}],9:[function(require,module,exports){
"use strict";
var deatultOptions = {
    request: {
        bindings: {
            activated: false,
            cache: false,
            constraint: false,
            dynamicValue: false,
            factory: false,
            implementationType: true,
            onActivation: false,
            provider: false,
            scope: true,
            serviceIdentifier: false,
            type: true
        },
        serviceIdentifier: true,
        target: {
            metadata: true,
            name: true,
            serviceIdentifier: true
        }
    },
    time: true
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = deatultOptions;

},{}],10:[function(require,module,exports){
"use strict";
function bindingTypeFormatter(type) {
    switch (type) {
        case 1:
            return "Instance";
        case 2:
            return "ConstantValue";
        case 3:
            return "DynamicValue";
        case 4:
            return "Constructor";
        case 5:
            return "Factory";
        case 6:
            return "Function";
        case 7:
            return "Provider";
        case 0:
            return "Invalid";
    }
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = bindingTypeFormatter;

},{}],11:[function(require,module,exports){
"use strict";
function scopeFormatter(scope) {
    switch (scope) {
        case 1:
            return "Singleton";
        case 0:
            return "Transient";
    }
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = scopeFormatter;

},{}],12:[function(require,module,exports){
"use strict";
var default_settings_1 = require("./config/default_settings");
var console_renderer_1 = require("./renderers/console_renderer");
var request_reducer_1 = require("./reducers/request_reducer");
var text_serializer_1 = require("./serializers/text/text_serializer");
exports.textSerializer = text_serializer_1.default;
var utils_1 = require("./utils/utils");
var utils_2 = require("./utils/utils");
var binding_type_formatter_1 = require("./formatters/binding_type_formatter");
exports.bindingTypeFormatter = binding_type_formatter_1.default;
var scope_formatter_1 = require("./formatters/scope_formatter");
exports.scopeFormatter = scope_formatter_1.default;
function makeLoggerMiddleware(settings, renderer) {
    var logger = function (planAndResolve) {
        return function (args) {
            if (settings === undefined || settings === null) {
                settings = default_settings_1.default;
            }
            ;
            if (renderer === undefined || renderer === null) {
                renderer = console_renderer_1.default;
            }
            ;
            var results = null;
            var logEntry = {
                error: false,
                exception: null,
                guid: utils_1.guid(),
                multiInject: args.multiInject,
                results: [],
                rootRequest: null,
                serviceIdentifier: args.serviceIdentifier,
                target: args.target,
                time: null
            };
            var nextContextInterceptor = args.contextInterceptor;
            args.contextInterceptor = function (context) {
                logEntry.rootRequest = request_reducer_1.default(context.plan.rootRequest, settings.request);
                return nextContextInterceptor(context);
            };
            try {
                var start = utils_2.getTime();
                results = planAndResolve(args);
                var end = utils_2.getTime();
                logEntry.results = results;
                logEntry.time = (settings.time) ? utils_2.getTimeDiference(start, end) : null;
            }
            catch (e) {
                logEntry.error = true;
                logEntry.exception = e;
                logEntry.time = null;
            }
            renderer(logEntry);
            return results || [];
        };
    };
    return logger;
}
exports.makeLoggerMiddleware = makeLoggerMiddleware;

},{"./config/default_settings":9,"./formatters/binding_type_formatter":10,"./formatters/scope_formatter":11,"./reducers/request_reducer":14,"./renderers/console_renderer":16,"./serializers/text/text_serializer":21,"./utils/utils":23}],13:[function(require,module,exports){
"use strict";
function bindingReducer(binding, options) {
    var props = [
        "type", "serviceIdentifier", "implementationType",
        "activated", "cache", "constraint", "dynamicValue",
        "factory", "onActivation", "provider", "scope"
    ];
    var reducedBinding = {};
    var bindingOptions = options;
    var oldBinding = binding;
    props.forEach(function (prop) {
        if (bindingOptions[prop]) {
            reducedBinding[prop] = oldBinding[prop];
        }
    });
    return reducedBinding;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = bindingReducer;

},{}],14:[function(require,module,exports){
"use strict";
var binding_reducer_1 = require("./binding_reducer");
var target_reducer_1 = require("./target_reducer");
function requestReducer(request, options) {
    var reducedRequest = {};
    var requestOptions = options;
    if (requestOptions.serviceIdentifier === true) {
        reducedRequest.serviceIdentifier = request.serviceIdentifier;
    }
    if (requestOptions.bindings !== undefined) {
        var reducedBindings = request.bindings.map(function (binding) {
            return binding_reducer_1.default(binding, options.bindings);
        });
        reducedRequest.bindings = reducedBindings;
    }
    if (requestOptions.target !== undefined && request.target !== null) {
        reducedRequest.target = target_reducer_1.default(request.target, options.target);
    }
    var reducedChieldRequest = request.childRequests.map(function (childRequest) {
        return requestReducer(childRequest, options);
    });
    reducedRequest.childRequests = reducedChieldRequest;
    return reducedRequest;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = requestReducer;

},{"./binding_reducer":13,"./target_reducer":15}],15:[function(require,module,exports){
"use strict";
function targetReducer(target, options) {
    var reducedTarget = {};
    var targetOptions = options;
    var oldTarget = target;
    if (targetOptions.name) {
        reducedTarget.name = oldTarget.name;
    }
    if (targetOptions.serviceIdentifier) {
        reducedTarget.serviceIdentifier = oldTarget.serviceIdentifier;
    }
    if (targetOptions.metadata && Array.isArray(target.metadata)) {
        var reducedMetadata = target.metadata.map(function (m, i) {
            return {
                key: m.key,
                value: m.value
            };
        });
        reducedTarget.metadata = reducedMetadata;
    }
    return reducedTarget;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = targetReducer;

},{}],16:[function(require,module,exports){
"use strict";
var text_serializer_1 = require("../serializers/text/text_serializer");
function consoleRenderer(entry) {
    return text_serializer_1.default(entry);
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = consoleRenderer;

},{"../serializers/text/text_serializer":21}],17:[function(require,module,exports){
"use strict";
var text_serializer_utils_1 = require("./text_serializer_utils");
var binding_type_formatter_1 = require("../../formatters/binding_type_formatter");
var scope_formatter_1 = require("../../formatters/scope_formatter");
function serializeBinding(textEntry, depth, binding) {
    var indentation = text_serializer_utils_1.getIndentationForDepth(depth);
    var propertyLogger = text_serializer_utils_1.makePropertyLogger(indentation);
    var props = [
        "type", "serviceIdentifier", "implementationType",
        "activated", "cache", "constraint", "dynamicValue",
        "factory", "onActivation", "provider", "scope"
    ];
    var _b = binding;
    props.forEach(function (prop) {
        if (_b[prop] !== undefined) {
            var val = _b[prop];
            switch (prop) {
                case "type":
                    val = binding_type_formatter_1.default(_b[prop]);
                    break;
                case "scope":
                    val = scope_formatter_1.default(_b[prop]);
                    break;
                case "implementationType":
                    val = _b[prop].name;
                    break;
                default:
                    val = _b[prop];
            }
            textEntry = propertyLogger(textEntry, 3, prop, val);
        }
    });
    return textEntry;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = serializeBinding;

},{"../../formatters/binding_type_formatter":10,"../../formatters/scope_formatter":11,"./text_serializer_utils":22}],18:[function(require,module,exports){
"use strict";
var text_serializer_utils_1 = require("./text_serializer_utils");
function serializeMetadata(textEntry, depth, metadatas) {
    var indentation = text_serializer_utils_1.getIndentationForDepth(depth);
    var propertyLogger = text_serializer_utils_1.makePropertyLogger(indentation);
    if (metadatas && Array.isArray(metadatas)) {
        textEntry = propertyLogger(textEntry, 2, "metadata");
        metadatas.forEach(function (metadata, index) {
            textEntry = propertyLogger(textEntry, 3, "Metadata", index.toString());
            textEntry = propertyLogger(textEntry, 4, "key", metadata.key);
            textEntry = propertyLogger(textEntry, 4, "value", metadata.value);
        });
    }
    return textEntry;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = serializeMetadata;

},{"./text_serializer_utils":22}],19:[function(require,module,exports){
"use strict";
var text_serializer_utils_1 = require("./text_serializer_utils");
var binding_serializer_1 = require("./binding_serializer");
var target_serializer_1 = require("./target_serializer");
function serializeRequest(textEntry, depth, index, request) {
    var indentation = text_serializer_utils_1.getIndentationForDepth(depth);
    var propertyLogger = text_serializer_utils_1.makePropertyLogger(indentation);
    textEntry = propertyLogger(textEntry, 0, "Request", index.toString());
    if (request.serviceIdentifier !== undefined) {
        textEntry = propertyLogger(textEntry, 1, "serviceIdentifier", request.serviceIdentifier);
    }
    if (request.bindings !== undefined && Array.isArray(request.bindings)) {
        textEntry = propertyLogger(textEntry, 1, "bindings");
        request.bindings.forEach(function (binding, bindingIndex) {
            textEntry = propertyLogger(textEntry, 2, "Binding<" + request.serviceIdentifier + ">", bindingIndex.toString());
            textEntry = binding_serializer_1.default(textEntry, depth, binding);
        });
    }
    textEntry = target_serializer_1.default(textEntry, depth, request.target);
    if (request.childRequests !== undefined && Array.isArray(request.childRequests) && request.childRequests.length > 0) {
        textEntry = propertyLogger(textEntry, 1, "childRequests");
        request.childRequests.forEach(function (childRequest, childIndex) {
            textEntry = serializeRequest(textEntry, (depth + 2), childIndex, childRequest);
        });
    }
    return textEntry;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = serializeRequest;

},{"./binding_serializer":17,"./target_serializer":20,"./text_serializer_utils":22}],20:[function(require,module,exports){
"use strict";
var metadata_serializer_1 = require("./metadata_serializer");
var text_serializer_utils_1 = require("./text_serializer_utils");
function serializeTarget(textEntry, depth, target) {
    var indentation = text_serializer_utils_1.getIndentationForDepth(depth);
    var propertyLogger = text_serializer_utils_1.makePropertyLogger(indentation);
    if (target !== undefined) {
        textEntry = propertyLogger(textEntry, 1, "target");
        if (target.serviceIdentifier !== undefined) {
            textEntry = propertyLogger(textEntry, 2, "serviceIdentifier", target.serviceIdentifier);
        }
        if (target.name !== undefined) {
            textEntry = propertyLogger(textEntry, 2, "name", (target.name.value() || "undefined"));
        }
        textEntry = metadata_serializer_1.default(textEntry, depth, target.metadata);
    }
    return textEntry;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = serializeTarget;

},{"./metadata_serializer":18,"./text_serializer_utils":22}],21:[function(require,module,exports){
"use strict";
var request_serializer_1 = require("./request_serializer");
var text_serializer_utils_1 = require("./text_serializer_utils");
function textSerializer(entry) {
    var textEntry = "";
    if (entry.error) {
        textEntry = textEntry + "\n" + text_serializer_utils_1.red("\nERROR: " + entry.exception.message + "\n" + entry.exception.stack) + "\n";
    }
    else {
        textEntry = textEntry + "\n" + text_serializer_utils_1.green("SUCCESS: " + entry.time + " ms.") + "\n";
    }
    textEntry = request_serializer_1.default(textEntry, 0, 0, entry.rootRequest);
    return textEntry;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = textSerializer;

},{"./request_serializer":19,"./text_serializer_utils":22}],22:[function(require,module,exports){
"use strict";
var chalk = require("chalk");
var tree = {
    item: "└──"
};
var indentationStr = "    ";
var yellow = chalk.yellow;
exports.yellow = yellow;
var green = chalk.green;
exports.green = green;
var red = chalk.red;
exports.red = red;
function getIndentationForDepth(depth) {
    var indentationForDepth = indentationStr;
    for (var i = depth; i--; i > 0) {
        indentationForDepth = "" + indentationForDepth + indentationStr;
    }
    return indentationForDepth;
}
exports.getIndentationForDepth = getIndentationForDepth;
function makePropertyLogger(indentationForDepth) {
    return function (textEntry, tabCount, key, value) {
        var line = "" + textEntry + indentationForDepth;
        for (var i = tabCount; i > 0; i--) {
            line = "" + line + indentationStr;
        }
        line = "" + line + tree.item + " " + key;
        if (value !== undefined) {
            if (typeof value !== "boolean") {
                var val = value ? value.toString().split("\n").join("") : "null";
                line = line + " : " + yellow(val);
            }
            else {
                line = line + " : " + yellow(value.toString());
            }
        }
        return line + "\n";
    };
}
exports.makePropertyLogger = makePropertyLogger;

},{"chalk":3}],23:[function(require,module,exports){
(function (process){
"use strict";
function getTime() {
    if (typeof window !== "undefined") {
        return window.performance.now();
    }
    else if (typeof process !== "undefined") {
        var nanoseconds = process.hrtime()[1];
        var milliseconds = nanoseconds / 1000000;
        return milliseconds;
    }
    else {
        return new Date().getTime();
    }
}
exports.getTime = getTime;
function guid() {
    function s4() {
        return Math.floor((1 + Math.random()) * 0x10000)
            .toString(16)
            .substring(1);
    }
    return s4() + s4() + "-" + s4() + "-" + s4() + "-" +
        s4() + "-" + s4() + s4() + s4();
}
exports.guid = guid;
function getTimeDiference(start, end) {
    var diff = end - start;
    var formatted = diff.toFixed(2);
    return formatted;
}
exports.getTimeDiference = getTimeDiference;

}).call(this,require('_process'))

},{"_process":6}]},{},[12])(12)
});
