// inversify-devtools/lib/containers/pages/settings_page.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var React = require("react");
var redux_1 = require("redux");
var react_redux_1 = require("react-redux");
var logger_actions_1 = require("../../actions/logger_actions");
var settings_actions_1 = require("../../actions/settings_actions");
var settings_1 = require("../../components/settings");
var utils_1 = require("../../utils/utils");
var actions = utils_1.combineActionsGroups(logger_actions_1.default, settings_actions_1.default);
function mapStateToPropsReposPage(state) {
    return {
        app: state.get("app"),
        settings: state.get("settings")
    };
}
function mapDispatchToPropsReposPage(dispatch) {
    return { actions: redux_1.bindActionCreators(actions, dispatch) };
}
var SettingsPage = (function (_super) {
    __extends(SettingsPage, _super);
    function SettingsPage() {
        _super.apply(this, arguments);
    }
    SettingsPage.prototype.render = function () {
        return (React.createElement(settings_1.default, {height: this.props.app.get("windowHeight"), columnSize: 12, settings: this.props.settings.get("settings"), saveSettingsAsync: this.props.actions.saveSettingsAsync.bind(this)}));
    };
    return SettingsPage;
}(React.Component));
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = react_redux_1.connect(mapStateToPropsReposPage, mapDispatchToPropsReposPage)(SettingsPage);
